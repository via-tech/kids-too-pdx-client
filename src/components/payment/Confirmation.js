import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Confirmation({ subFee, confirmation }) {
  if(confirmation) return (
    <section>
      <p>Payment successful! Subscription will automatically renew each year. If on trial period, a ${subFee} subscription fee will automatically be charged after trial ends. Email reminders will be sent out accordingly. To manage subscription preferences, please visit the dashboard. Thank you!</p>

      <p>
        Continue to
        <Link to="/">Home</Link> page
      </p>
    </section>
  );

  return <p>
    Confirmation page only upon payment!
    Go <Link to="/">Home</Link>
  </p>;
}

Confirmation.propTypes = {
  subFee: PropTypes.string.isRequired,
  confirmation: PropTypes.bool.isRequired
};

export default Confirmation;
