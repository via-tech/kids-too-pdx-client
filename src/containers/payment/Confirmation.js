import { connect } from 'react-redux';
import React from 'react';
import ConfirmationPage from '../../components/payment/ConfirmationPage';
import { getConfirmation } from '../../selectors/session';

function Confirmation(props) {
  return <ConfirmationPage {...props} />;
}

const mapStateToProps = state => ({
  subFee: process.env.SUB_FEE_PROMO,
  confirmation: getConfirmation(state)
});

export default connect(
  mapStateToProps
)(Confirmation);
