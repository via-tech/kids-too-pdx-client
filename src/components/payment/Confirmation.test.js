import React from 'react';
import renderer from 'react-test-renderer';
import Confirmation from './Confirmation';
import { MemoryRouter } from 'react-router-dom';

describe('Confirmation', () => {
  it('snapshots confirmation page', () => {
    const props = {
      subFee: 500,
      confirmation: true
    };

    const tree = renderer.create(
      <MemoryRouter>
        <Confirmation {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots confirmation page without confirmation', () => {
    const props = {
      subFee: 500,
      confirmation: false
    };

    const tree = renderer.create(
      <MemoryRouter>
        <Confirmation {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
