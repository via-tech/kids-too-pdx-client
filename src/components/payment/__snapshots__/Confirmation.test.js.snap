// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation snapshots confirmation page 1`] = `
<section>
  <p>
    Payment successful! Subscription will automatically renew each year. If on trial period, a $
    500
     subscription fee will automatically be charged after trial ends. Email reminders will be sent out accordingly. To manage subscription preferences, please visit the dashboard. Thank you!
  </p>
  <p>
    Continue to
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
     page
  </p>
</section>
`;

exports[`Confirmation snapshots confirmation page without confirmation 1`] = `
<p>
  Confirmation page only upon payment! Go 
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</p>
`;
